tests/loader/test_astronet_loader.py

```diff
@@ -0,0 +1,164 @@
+
+    def test_get_image_invalid_index(self):
+        """Test that get_image raises an exception on an invalid index."""
+        # negative index
+        with self.assertRaises(IndexError):
+            self.loader.get_image(-1)
+        # len() as index
+        with self.assertRaises(IndexError):
+            self.loader.get_image(len(self.loader))
+        # index > len()
+        with self.assertRaises(IndexError):
+            self.loader.get_image(25)
+
+    def test_get_camera_intrinsics(self) -> None:
+        """Ensure that for shared calibration case, GT intrinsics are identical across frames."""
+        # TODO: also test case with multiple calibrations
+        K0 = self.loader.get_camera_intrinsics(0).K()
+        K1 = self.loader.get_camera_intrinsics(1).K()
+        np.testing.assert_allclose(K0, K1)
+
+    def test_get_image(self) -> None:
+        """Ensure an image can be successfully provided."""
+        img0 = self.loader.get_image(0)
+        assert isinstance(img0, Image)
+
+    def test_get_sfmtrack_valid_index(self):
+        """Tests that get_sfmtrack works for all valid indices."""
+        for idx_sfmtrack in TEST_SFMTRACKS_INDICES:
+            sfmtrack = self.loader.get_sfmtrack(idx_sfmtrack)
+            self.assertIsNotNone(sfmtrack.point3())
+
+    def test_get_sfmtrack_invalid_index(self):
+        """Test that get_sfmtrack raises an exception on an invalid index."""
+        # negative index
+        with self.assertRaises(IndexError):
+            self.loader.get_sfmtrack(-1)
+        # len() as index
+        with self.assertRaises(IndexError):
+            self.loader.get_sfmtrack(self.loader.num_sfmtracks)
+        # index > len()
+        with self.assertRaises(IndexError):
+            self.loader.get_sfmtrack(99999)
+
+    def test_image_contents(self):
+        """Test the actual image which is being fetched by the loader at an index.
+
+        This test's primary purpose is to check if the ordering of filename is being respected by the loader
+        """
+        index_to_test = 1
+        file_path = TEST_DATA_ROOT / "test_2011212_opnav_022" / "images" / "00000001.png"
+        loader_image = self.loader.get_image(index_to_test)
+        expected_image = io_utils.load_image(file_path)
+        np.testing.assert_allclose(expected_image.value_array, loader_image.value_array)
+
+    def test_get_camera_pose_exists(self):
+        """Tests that the correct pose is fetched (present on disk)."""
+        fetched_pose = self.loader.get_camera_pose(1)
+        self.assertTrue(isinstance(fetched_pose, Pose3))
+
+    def test_get_camera_invalid_index(self):
+        """Tests that the camera pose is None, because it is missing on disk."""
+        with self.assertRaises(IndexError):
+            fetched_pose = self.loader.get_camera_pose(25)
+            self.assertIsNone(fetched_pose)
+
+    def test_trimesh_sfmtracks_align(self):
+        """Tests that the ground truth scene mesh aligns with sfmtracks"""
```

Comment:
could you explain a bit more in the docstring about what it means to align a mesh with an SfmTrack?

Reply:
Changed this unit test to not use the ground truth mesh (yet) and updated docstring.